pycodestyle.py

```diff
@@ -1995,7 +1995,7 @@ def __init__(self, *args, **kwargs):
         # build options from dict
         options_dict = dict(*args, **kwargs)
         arglist = None if parse_argv else options_dict.get('paths', None)
-        verbose = options_dict.get('verbose', 0)
+        verbose = options_dict.get('verbose', None)
```

Comment:
Please use `dict.pop`.

Reply:
`pop` will not work because at the end of `read_config`, `options` is overwritten with command line arguments, which means if `-v` was not passed somehow it would default to 0. 

https://github.com/PyCQA/pycodestyle/blob/6faef2bfda5752c87d432872fe0000aa13316014/pycodestyle.py#L2303-L2304

This problem has not been apparent because as you pointed out earlier, `options_dict` gets copied to `options.__dict__`, but if I pop the `verbose` key, will not get copied.

Alternatively, `(options, __) = parser.parse_args(arglist, values=options)` seems to work, but I'm not sure why the previous `values=new_options` was done the way it was.

Reply:
Thanks for pointing that out. I missed that.

So if you look at the loop that builds up `new_options` it's pulling in values from config files. But in the chain of preferences, we always want CLI arguments to override those. This is why we one last time do `parser.parse_args(arglist, values=new_options)`. We want to default to what's in the config file, and override those values with what's provided on the CLI.